src/main/java/ratpack/zipkin/internal/RatpackCurrentTraceContext.java

```diff
@@ -21,26 +25,47 @@ public RatpackCurrentTraceContext() {
   @Override
   public TraceContext get() {
     return registrySupplier.get()
-        .maybeGet(TraceContextHolder.class)
-        .map(TraceContextHolder::getContext)
+        .maybeGet(ContextHolder.class)
+        .map(h -> h.context)
         .orElse(null);
   }
 
   @Override
   public Scope newScope(TraceContext currentSpan) {
-    final TraceContext previous = get();
-    registrySupplier.get().add(TraceContextHolder.class, new TraceContextHolder(currentSpan));
-    return () -> registrySupplier.get().add(new TraceContextHolder(previous));
+
+    if (currentSpan != null) {
+
+      final Optional<ContextHolder> previous = registrySupplier.get()
+          .maybeGet(ContextHolder.class);
+
+      registrySupplier.get().add(new ContextHolder(currentSpan));
+      MDC.put(MCD_TRACEID_KEY, currentSpan.traceIdString());
+
+      return () -> {
+        if (previous.isPresent() && previous.get().context != null) {
+          registrySupplier.get().add(previous);
+          MDC.put(MCD_TRACEID_KEY, previous.get().context.traceIdString());
```

Comment:
We could probably make the MDC stuff optional if others feel its not always desirable.  I thought I would just throw it in here and get feedback first before doing extra work to make it optional :)

Reply:
@llinder personally, I'm a fan of MDC, so I'm not opposed to having it in. But if it's not too tied into the upgrade to V2 spans, you could just split it out into a separate PR and we can work on getting this one merged before tackling that.

Reply:
Sure I will make it a different PR.  Actually that part is what got me started on this because I need something like that sooner than the V2 stuff.  Figured I would try and tackle the V2 stuff as long as I was in the code though.